chia/full_node/bitcoin_fee_estimator.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import logging
+from datetime import datetime
+from typing import Any, Dict
+
+from chia.full_node.fee_estimate_store import FeeStore
+from chia.full_node.fee_estimation import FeeBlockInfo, FeeMempoolInfo
+from chia.full_node.fee_estimator import SmartFeeEstimator
+from chia.full_node.fee_estimator_interface import FeeEstimatorInterface
+from chia.full_node.fee_tracker import FeeTracker
+from chia.types.clvm_cost import CLVMCost
+from chia.types.fee_rate import FeeRate
+from chia.types.mempool_item import MempoolItem
+from chia.types.mojos import Mojos
+from chia.util.ints import uint32, uint64
+
+MIN_MOJO_PER_COST = 5
+
+
+def demo_fee_rate_function(cost: int, time_in_seconds: int) -> uint64:
+    return uint64(cost * MIN_MOJO_PER_COST * max((3600 - time_in_seconds), 1))
+
+
+class BitcoinFeeEstimator(FeeEstimatorInterface):
+    """
+    A Fee Estimator based on the concepts and code at:
+    https://github.com/bitcoin/bitcoin/tree/5b6f0f31fa6ce85db3fb7f9823b1bbb06161ae32/src/policy
+    """
+
+    def __init__(self, config: Dict[str, Any]) -> None:
+        self.fee_rate_estimator: SmartFeeEstimator = config["estimator"]
+        self.tracker: FeeTracker = config["tracker"]
+        self.last_mempool_info = FeeMempoolInfo(
+            CLVMCost(uint64(0)),
+            FeeRate.create(Mojos(uint64(0)), CLVMCost(uint64(1))),
+            CLVMCost(uint64(0)),
+            datetime.min,
+            CLVMCost(uint64(0)),
+        )
+
+    def new_block(self, block_info: FeeBlockInfo) -> None:
+        self.tracker.process_block(block_info.block_height, block_info.included_items)
+
+    def add_mempool_item(self, mempool_info: FeeMempoolInfo, mempool_item: MempoolItem) -> None:
+        self.last_mempool_info = mempool_info
+
+    def remove_mempool_item(self, mempool_info: FeeMempoolInfo, mempool_item: MempoolItem) -> None:
```

Comment:
is it ok that we dont actually remove items that get removed from the actual mempool ?

seems like this can effect mempool_size that would change estimations